File: Day1/287-Find the-Duplicate-Number/Method1.cpp
class Solution
{
public:
    int findDuplicate(vector<int> &nums)
    {
        unordered_set<int> st;

        int n = nums.size();
        for (int i = 0; i < n; i++)
        {
            if (st.count(nums[i]))
                return nums[i];
            st.insert(nums[i]);
        }
        return -1;
    }
};

File: Day2/724-Find-Pivot-Index/Method1.java
class Solution {
    public int pivotIndex(int[] nums) {
        int n = nums.length;
        int[] leftSum = new int[n];
        int[] rightSum = new int[n];

        for (int i = 0; i < n; i++) {
            if (i == 0) {
                leftSum[i] = nums[i];
                continue;
            }
            leftSum[i] = leftSum[i - 1] + nums[i];
        }

        for (int i = n - 1; i >= 0; i--) {
            if (i == n - 1) {
                rightSum[i] = nums[i];
                continue;
            }
            rightSum[i] = rightSum[i + 1] + nums[i];
        }

        for (int i = 0; i < n; i++) {
            if (leftSum[i] == rightSum[i])
                return i;
        }

        return -1;
    }
}